### PR TITLE
DA v2: fix glossaries permissions and refactor catalog module

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix
- Refactoring

### Detail
- moved glossaries permissions from core to module.catalog
- refractored catalog module to follow services, resolvers, db layer design
- fix list_term_associations
- remove Columns from glossaries registry --> they cannot be tagged
- clean up unused code in glossaries

### Relates
V2.0.0 release

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

`N/A` 
- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Still testing:
- [ ] locally
- [ ] on AWS

**Comment by dlpzx** [comment]
I noticed that there is an error with the `list_term_associations` database query in V2. 

Despite having associations (which I checked in the database), the associations view shows no associations:
image

The issue is in the changes in this query:

Original in V1.6

```
@staticmethod
    def list_term_associations(
        session, username, groups, uri, data=None, check_perm=None
    ):
        source = data['source']
        filter = data['filter']
        datasets = session.query(
            models.Dataset.datasetUri.label('targetUri'),
            literal('dataset').label('targetType'),
            models.Dataset.label.label('label'),
            models.Dataset.name.label('name'),
            models.Dataset.description.label('description'),
        )
        tables = session.query(
            models.DatasetTable.tableUri.label('targetUri'),
            literal('table').label('targetType'),
            models.DatasetTable.label.label('label'),
            models.DatasetTable.name.label('name'),
            models.DatasetTable.description.label('description'),
        )
        columns = session.query(
            models.DatasetTableColumn.columnUri.label('targetUri'),
            literal('column').label('targetType'),
            models.DatasetTableColumn.label.label('label'),
            models.DatasetTableColumn.name.label('name'),
            models.DatasetTableColumn.description.label('description'),
        )
        folders = session.query(
            models.DatasetStorageLocation.locationUri.label('targetUri'),
            literal('folder').label('targetType'),
            models.DatasetStorageLocation.label.label('label'),
            models.DatasetStorageLocation.name.label('name'),
            models.DatasetStorageLocation.description.label('description'),
        )

        dashboards = session.query(
            models.Dashboard.dashboardUri.label('targetUri'),
            literal('dashboard').label('targetType'),
            models.Dashboard.label.label('label'),
            models.Dashboard.name.label('name'),
            models.Dashboard.description.label('description'),
        )

        linked_objects = datasets.union(tables, columns, folders, dashboards).subquery(
            'linked_objects'
        )

        path = models.GlossaryNode.path
        q = (
            session.query(models.TermLink)
            .options(with_expression(models.TermLink.path, path))
            .join(
                models.GlossaryNode,
                models.GlossaryNode.nodeUri == models.TermLink.nodeUri,
            )
            .join(
                linked_objects, models.TermLink.targetUri == linked_objects.c.targetUri
            )
        )
        if source.nodeType == 'T':
            q = q.filter(models.TermLink.nodeUri == source.nodeUri)
        elif source.nodeType in ['C', 'G']:
            q = q.filter(models.GlossaryNode.path.startswith(source.path))
        else:
            raise Exception(f'InvalidNodeType ({source.nodeUri}/{source.nodeType})')

        term = filter.get('term')
        if term:
            q = q.filter(
                or_(
                    linked_objects.c.label.ilike('%' + term + '%'),
                    linked_objects.c.description.ilike(f'%{term}'),
                    linked_objects.c.targetType.ilike(f'%{term}'),
                )
            )
        q = q.order_by(asc(path))

        return paginate(
            q, page=filter.get('page', 1), page_size=filter.get('pageSize', 25)
        ).to_dict()
```

New one in V2 (without the print statements)

```
@staticmethod
    def list_term_associations(session, target_model_definitions, node, filter=None):
        print("repo")
        query = None
        for definition in target_model_definitions:
            model = definition.model
            subquery = session.query(
                definition.target_uri().label('targetUri'),
                literal(definition.target_type.lower()).label('targetType'),
                model.label.label('label'),
                model.name.label('name'),
                model.description.label('description'),
            )
            if query:
                query.union(subquery)
            else:
                query = subquery

        if query is None:
            return Page([], 1, 1, 0)  # empty page. All modules are turned off

        linked_objects = query.subquery('linked_objects')

        path = GlossaryNode.path
        print(f"path: {path}")
        q = (
            session.query(TermLink)
            .options(with_expression(TermLink.path, path))
            .join(
                GlossaryNode,
                GlossaryNode.nodeUri == TermLink.nodeUri,
            )
            .join(
                linked_objects, TermLink.targetUri == linked_objects.c.targetUri
            )
        )
        for item in q:
            print("query up")
            print(item)
        if node.nodeType == 'T':
            print("is term")
            q = q.filter(TermLink.nodeUri == node.nodeUri)
        elif node.nodeType in ['C', 'G']:
            print("is category/glos")
            print(node.path)
            q = q.filter(GlossaryNode.path.startswith(node.path))
        else:
            raise Exception(f'InvalidNodeType ({node.nodeUri}/{node.nodeType})')

        term = filter.get('term')
        if term:
            q = q.filter(
                or_(
                    linked_objects.c.label.ilike('%' + term + '%'),
                    linked_objects.c.description.ilike(f'%{term}'),
                    linked_objects.c.targetType.ilike(f'%{term}'),
                )
            )
        q = q.order_by(asc(path))

        print("query")
        print(q)
        for item in q:
            print(item)

        return paginate(
            q, page=filter.get('page', 1), page_size=filter.get('pageSize', 25)
        ).to_dict()
```

File: frontend/docker/dev/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ COPY ./frontend/yarn.lock ./
 
 # Install packages, use --ignore-scripts to not call postinstall, as it causes this step to fail because config.json
 # is not copied yet, and copying it here will trigger new install if config changes, which is inconvenient for development.
-RUN yarn install --ignore-scripts
+RUN yarn install --ignore-scripts --verbose --network-timeout 100000
```
Comment (nikpodsh) [inline frontend/docker/dev/Dockerfile:10]: Was this change intentional ?

Comment (dlpzx) [reply frontend/docker/dev/Dockerfile]: Nope, I am removing it in this [commit](https://github.com/awslabs/aws-dataall/commit/ceefd340f3cf438eac1a6580ced1934611a3ade3), plus I added some logs for the graphql calls. That is intentional, a customer requested those

**Comment by dlpzx** [comment]
@nikpodsh I also found the issue with the "invisible" associations. It is fixed in this [commit](https://github.com/awslabs/aws-dataall/commit/fdacac1e352a5d17966c87e1f94cca68352bd16f). Everything works for me locally: creation and deletion of glossaries, categories and terms. Listing glossaries and its tree of categories and terms. Listing associations and approving and dismissing them.

**Comment by nikpodsh** [comment]
I am still observing empty associations: 
Screenshot 2023-09-07 at 14 02 33
+ filtering by glossaries in catalog is not working for me. Checking

**Comment by dlpzx** [comment]
> I am still observing empty associations: Screenshot 2023-09-07 at 14 02 33
> 
>     * filtering by glossaries in catalog is not working for me. Checking

Yes, I can confirm that for me the `list_associations` also does not work in AWS

**Review by nikpodsh — APPROVED** [review]
Tested it in my account. It's working perfectly ! 
@dlpzx , thanks for your work 🚀
